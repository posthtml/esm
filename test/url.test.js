const posthtml = require('./helpers')
const { urls } = require('../lib')

describe('URLs', () => {
  describe('Defaults', () => {
    test('src', () => {
      const plugins = [ urls() ]
      const options = {}

      return posthtml('url/src.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })

    test('href', () => {
      const plugins = [ urls() ]
      const options = {}

      return posthtml('url/href.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })

    test('srcset', () => {
      const plugins = [ urls() ]
      const options = {}

      return posthtml('url/srcset.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })
  })

  describe('Filter', () => {
    test('src', () => {
      const plugins = [
        urls({
          url: /filter/
        })
      ]
      const options = {}

      return posthtml('url/src.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })

    test('href', () => {
      const plugins = [
        urls({
          url: /filter/
        })
      ]
      const options = {}

      return posthtml('url/href.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })

    test('srcset', () => {
      const plugins = [
        urls({
          filter: /filter/
        })
      ]
      const options = {}

      return posthtml('url/srcset.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()

          return null
        })
    })
  })
})
