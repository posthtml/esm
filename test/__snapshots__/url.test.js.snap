// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLs Defaults href 1`] = `
"<link href=\\"\${HTML__URL__0}\\">
<link href=\\"\${HTML__URL__1}\\">
"
`;

exports[`URLs Defaults href 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.css",
  },
]
`;

exports[`URLs Defaults src 1`] = `
"<img src=\\"\${HTML__URL__0}\\">
<img src=\\"\${HTML__URL__1}\\">
"
`;

exports[`URLs Defaults src 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
]
`;

exports[`URLs Defaults srcset 1`] = `
"​<picture>
  <source srcset=\\"\${HTML__URL__0}\\" type=\\"image/svg+xml\\">
  <img src=\\"\${HTML__URL__1}\\" alt=\\"Image\\">
</picture>

​<picture>
  <source srcset=\\"\${HTML__URL__2}\\" type=\\"image/svg+xml\\">
  <img src=\\"\${HTML__URL__3}\\" alt=\\"Image\\">
</picture>
"
`;

exports[`URLs Defaults srcset 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__2",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__3",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
]
`;

exports[`URLs Filter href 1`] = `
"<link href=\\"\${HTML__URL__0}\\">
<link href=\\"./filter/file.css\\">
"
`;

exports[`URLs Filter href 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__2",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__3",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
]
`;

exports[`URLs Filter src 1`] = `
"<img src=\\"\${HTML__URL__0}\\">
<img src=\\"./filter/file.png\\">
"
`;

exports[`URLs Filter src 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__2",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__3",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
]
`;

exports[`URLs Filter srcset 1`] = `
"​<picture>
  <source srcset=\\"\${HTML__URL__0}\\" type=\\"image/svg+xml\\">
  <img src=\\"\${HTML__URL__1}\\" alt=\\"Image\\">
</picture>

​<picture>
  <source srcset=\\"\${HTML__URL__2}\\" type=\\"image/svg+xml\\">
  <img src=\\"\${HTML__URL__3}\\" alt=\\"Image\\">
</picture>
"
`;

exports[`URLs Filter srcset 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__2",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__3",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.css",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__0",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./file.png",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__2",
    "plugin": "@posthtml/esm (URL)",
    "type": "import",
    "url": "./filter/file.svg",
  },
  Object {
    "import": [Function],
    "name": "HTML__URL__3",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/file.png",
  },
]
`;
